tests/PHPStan/Analyser/nsrt/preg_match_shapes.php

```diff
@@ -472,3 +470,21 @@ function (string $s): void {
 		assertType("array{string, non-empty-string}", $matches);
 	}
 };
+
+function bug11323(string $s): void {
+	if (preg_match('/([*|+?{}()]+)([^*|+[:digit:]?{}()]+)/', $s, $matches)) {
+		assertType('array{string, non-empty-string, non-empty-string}', $matches);
+	}
+	if (preg_match('/\p{L}[[\]]+([-*|+?{}(?:)]+)([^*|+[:digit:]?{a-z}(\p{L})\a-]+)/', $s, $matches, PREG_UNMATCHED_AS_NULL)) {
```

Comment:
I usually put tests which use `PREG_UNMATCHED_AS_NULL` into `tests/PHPStan/Analyser/nsrt/bug-11311.php` for assertions which only work with php74+

additionally it would make sense to put the same test with different assertions (for php < 7.4) into `tests/PHPStan/Analyser/nsrt/bug-11311-php72.php`

Reply:
I would also expect a test with just `(\p{L})` capturing, to verify we have the proper type for this regex character class inferred

Reply:
I just removed the PREG_UNMATCHED_AS_NULL as it's completely irrelevant, just landed there through copy paste. Will add a few new tests according to your feedback 👍🏻